dist/hyperlist.js

```diff
@@ -210,57 +250,63 @@ var HyperList = function () {
       var oldClass = item.getAttribute('class') || '';
       item.setAttribute('class', oldClass + ' ' + (config.rowClassName || 'vrow'));
 
-      var offsetTop = i * itemHeight;
-      var top = reverse ? (total - 1) * itemHeight - offsetTop : offsetTop;
+      var top = this._itemPositions[i];
 
-      item.setAttribute('style', '\n      ' + (item.style.cssText || '') + '\n      position: absolute;\n      top: ' + top + 'px\n    ');
+      this._mergeStyle(item, {
+        position: 'absolute',
+        top: top + 'px'
+      });
 
       return item;
     }
   }, {
-    key: _getScrollPosition,
-    value: function value() {
-      var config = this[_config];
+    key: '_getScrollPosition',
+    value: function _getScrollPosition() {
+      var config = this._config;
 
       if (typeof config.overrideScrollPosition === 'function') {
         return config.overrideScrollPosition();
       }
 
-      return this[_element].scrollTop;
+      return this._element.scrollTop;
     }
   }, {
-    key: _renderChunk,
-    value: function value() {
-      var config = this[_config];
-      var element = this[_element];
-      var scrollTop = this[_getScrollPosition]();
-      var screenItemsLen = this[_screenItemsLen];
-      var getRow = this[_getRow].bind(this);
+    key: '_renderChunk',
+    value: function _renderChunk() {
+      var config = this._config;
+      var element = this._element;
+      var scrollTop = this._getScrollPosition();
       var total = config.total;
-      var itemHeight = config.itemHeight;
-      var estFrom = Math.floor(scrollTop / itemHeight) - screenItemsLen;
-      var from = estFrom > total ? total : estFrom < 0 ? 0 : estFrom;
-      var estTo = from + this[_cachedItemsLen];
-      var to = estTo > total ? total : estTo < 0 ? 0 : estTo;
+
+      var from = config.reverse ? this._getReverseFrom(scrollTop) : this._getFrom(scrollTop) - 1;
+      from = from < 0 ? 0 : from;
+
+      if (this._lastFrom === from) {
+        return false;
+      }
+
+      this._lastFrom = from;
+
+      var to = from + this._cachedItemsLen;
+      to = to > total ? total : to;
+      console.log(from, to);
```

Comment:
Just a heads up you should rebuild the main package in this rebased final commit to remove this `console.log`.